### ISSUE TITLE
[FEATURE] LXD/LXC support for lightway vm like linux full system

### ISSUE THREAD

**Issue by aperez900907** [harvester/harvester#2012]
The availability in cases of low profiles hardware to run lightway linux containers instead of full kvm like vms will be very appreciate to run aside docker.

**Comment by emilyastranova** [comment]
Curious on how this differs from integration with a Rancher instance? I'm also looking at hybrid VM/container support and we were just going to add a Rancher server.

**Comment by yasker** [comment]
Sorry, other hypervisors/container technologies are out of scope for Harvester at this moment.

**Comment by scog** [comment]
@aperez900907 I second this request. LXC is very lightweight and is presently supported on Proxmox (KVM) and ESXi. 

If Harvester had the ability to run LXC containers, that would further increase competition within the hypervisor market considering the underlying advantages Harvester has @ the storage layer.

CC: @yasker

**Comment by xeruf** [comment]
Yes, switching from Proxmox for full-on Kubernetes, the need to setup full VMs with own OS is kinda annoying in comparison to lightweight LXC

**Comment by onegreyonewhite** [comment]
It would actually be a great idea to allow for a more lightweight way to run containers in a hypervisor cluster. This would be a step towards developing Harvester not just as a layer for quickly deploying a Kubernetes cluster, but also as a standalone product or private cloud.

IMHO, for implementation, I'm missing:
1. Flexible networking (creating separate networks, routing between them, OSPF or BGP between clusters).
2. Verification of backup and snapshot integrity.
3. The ability to run lightweight containers (but not Docker images).

**Comment by Nilox42** [comment]
LXC would be a great addition to Harvester. Many people running Proxmox would probably switch if they could use LXC inside Harvester. Myself included. I am still looking for a great Hypervisor that can run Kubernetes/Docker LXC and KVM all at the same time and well managed.
